File: StrykerDG.FarmForge.Api/Controllers/DTO/Requests/NewSupplierDTO.cs
using StrykerDG.FarmForge.DataModel.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace StrykerDG.FarmForge.LocalApi.Controllers.DTO.Requests
{
    public class NewSupplierDTO
    {
        public Supplier Supplier { get; set; }
        public List<int> ProductIds { get; set; }
    }
}

File: StrykerDG.FarmForge.Client/farmforge_client/lib/models/dto/new_supplier_dto.dart
import 'package:farmforge_client/models/suppliers/supplier.dart';

class NewSupplierDTO {
  Supplier supplier;
  List<int> productIds;

  NewSupplierDTO({
    this.supplier,
    this.productIds
  });

  Map<String, dynamic> toMap() {
    Map<String, dynamic> supplier = this.supplier != null
      ? this.supplier.toMap()
      : null;

    return {
      'Supplier': supplier,
      'ProductIds': this.productIds
    };
  }
}